src/base/Query.sol

```diff
@@ -15,13 +15,50 @@ import {
 
 import { ZERO_TIMESTAMP } from "../Common.sol";
 
+import { ModuleType, ModuleTypesEnc, ModuleTypeLib } from "src/DataTypes.sol";
+
 /**
  * @title Query
  * @author rhinestone | zeroknots.eth, Konrad Kopp (@kopy-kat)
  * Implements EIP-7484 to query attestations stored in the registry.
  * @dev This contract is abstract and provides utility functions to query attestations.
  */
 abstract contract Query is IQuery {
+    using ModuleTypeLib for ModuleTypesEnc;
+
+    function check(
+        address module,
+        address attester,
+        ModuleType moduleType
```

Comment:
this is kind of an out there idea but we could also retain the 7484 interface (if eg we think module types are too opinionated) and add the type after the attester (kinda like 2771)

Reply:
just have two check functions

one with type one without.